cli: TestUnavailableZip failed

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13415873?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13415873?buildTab=artifacts#/) on master @ [473b327220354696bfd850aadfbce1f03ff5ab1f](https://github.com/cockroachdb/cockroach/commits/473b327220354696bfd850aadfbce1f03ff5ab1f):


```
        	            	  | runtime.panicmem
        	            	  | 	GOROOT/src/runtime/panic.go:261
        	            	  | runtime.sigpanic
        	            	  | 	GOROOT/src/runtime/signal_unix.go:861
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.(*debugZipContext).collectPerNodeData
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip_per_node.go:476
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip.func3.2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:305
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.(*debugZipContext).forAllNodes
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:107
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip.func3
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:304
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:347
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.init.MaybeDecorateError.func109
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/clierrorplus/decorate_error.go:72
        	            	  | github.com/spf13/cobra.(*Command).execute
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:856
        	            	  | github.com/spf13/cobra.(*Command).ExecuteC
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:974
        	            	  | github.com/spf13/cobra.(*Command).Execute
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:902
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/cli.go:304
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithArgs.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:433
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithArgs
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:434
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:291
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithCapture.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:300
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.captureOutput
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:352
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithCapture
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:299
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestUnavailableZip.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip_test.go:441
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1197
        	            	Wraps: (2) secondary error attachment
        	            	  | runtime error: invalid memory address or nil pointer dereference
        	            	  | (1) runtime error: invalid memory address or nil pointer dereference
        	            	  | Error types: (1) runtime.errorString
        	            	Wraps: (3) panic: runtime error: invalid memory address or nil pointer dereference
        	            	Error types: (1) *withstack.withStack (2) *secondary.withSecondaryError (3) *errutil.leafError
        	Test:       	TestUnavailableZip/server_1
    --- FAIL: TestUnavailableZip/server_1 (32.44s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-inf-prs @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35162

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13605141?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13605141?buildTab=artifacts#/) on master @ [eb42e39c887b9429cc46aed9db7ef66a4f6ebe82](https://github.com/cockroachdb/cockroach/commits/eb42e39c887b9429cc46aed9db7ef66a4f6ebe82):


```
        	            	  | runtime.panicmem
        	            	  | 	GOROOT/src/runtime/panic.go:261
        	            	  | runtime.sigpanic
        	            	  | 	GOROOT/src/runtime/signal_unix.go:861
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.(*debugZipContext).collectPerNodeData
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip_per_node.go:476
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip.func3.2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:305
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.(*debugZipContext).forAllNodes
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:107
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip.func3
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:304
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:347
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.init.MaybeDecorateError.func106
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/clierrorplus/decorate_error.go:72
        	            	  | github.com/spf13/cobra.(*Command).execute
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:856
        	            	  | github.com/spf13/cobra.(*Command).ExecuteC
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:974
        	            	  | github.com/spf13/cobra.(*Command).Execute
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:902
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/cli.go:303
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithArgs.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:433
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithArgs
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:434
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:291
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithCapture.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:300
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.captureOutput
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:352
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithCapture
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:299
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestUnavailableZip.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip_test.go:441
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1197
        	            	Wraps: (2) secondary error attachment
        	            	  | runtime error: invalid memory address or nil pointer dereference
        	            	  | (1) runtime error: invalid memory address or nil pointer dereference
        	            	  | Error types: (1) runtime.errorString
        	            	Wraps: (3) panic: runtime error: invalid memory address or nil pointer dereference
        	            	Error types: (1) *withstack.withStack (2) *secondary.withSecondaryError (3) *errutil.leafError
        	Test:       	TestUnavailableZip/server_1
    --- FAIL: TestUnavailableZip/server_1 (31.68s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13612341?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13612341?buildTab=artifacts#/) on master @ [0bff9dc86ef1a2e86c015c391c76f0b519d44cb4](https://github.com/cockroachdb/cockroach/commits/0bff9dc86ef1a2e86c015c391c76f0b519d44cb4):


```
        	            	  | runtime.panicmem
        	            	  | 	GOROOT/src/runtime/panic.go:261
        	            	  | runtime.sigpanic
        	            	  | 	GOROOT/src/runtime/signal_unix.go:861
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.(*debugZipContext).collectPerNodeData
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip_per_node.go:476
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip.func3.2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:305
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.(*debugZipContext).forAllNodes
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:107
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip.func3
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:304
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:347
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.init.MaybeDecorateError.func106
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/clierrorplus/decorate_error.go:72
        	            	  | github.com/spf13/cobra.(*Command).execute
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:856
        	            	  | github.com/spf13/cobra.(*Command).ExecuteC
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:974
        	            	  | github.com/spf13/cobra.(*Command).Execute
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:902
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/cli.go:303
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithArgs.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:433
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithArgs
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:434
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:291
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithCapture.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:300
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.captureOutput
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:352
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithCapture
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:299
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestUnavailableZip.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip_test.go:441
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1197
        	            	Wraps: (2) secondary error attachment
        	            	  | runtime error: invalid memory address or nil pointer dereference
        	            	  | (1) runtime error: invalid memory address or nil pointer dereference
        	            	  | Error types: (1) runtime.errorString
        	            	Wraps: (3) panic: runtime error: invalid memory address or nil pointer dereference
        	            	Error types: (1) *withstack.withStack (2) *secondary.withSecondaryError (3) *errutil.leafError
        	Test:       	TestUnavailableZip/server_1
    --- FAIL: TestUnavailableZip/server_1 (32.34s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13615965?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13615965?buildTab=artifacts#/) on master @ [2a4e6b81c11ff26cbb9a57b36f05b1e3c7589e3c](https://github.com/cockroachdb/cockroach/commits/2a4e6b81c11ff26cbb9a57b36f05b1e3c7589e3c):


```
        	            	  | runtime.panicmem
        	            	  | 	GOROOT/src/runtime/panic.go:261
        	            	  | runtime.sigpanic
        	            	  | 	GOROOT/src/runtime/signal_unix.go:861
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.(*debugZipContext).collectPerNodeData
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip_per_node.go:476
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip.func3.2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:305
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.(*debugZipContext).forAllNodes
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:107
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip.func3
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:304
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:347
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.init.MaybeDecorateError.func106
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/clierrorplus/decorate_error.go:72
        	            	  | github.com/spf13/cobra.(*Command).execute
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:856
        	            	  | github.com/spf13/cobra.(*Command).ExecuteC
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:974
        	            	  | github.com/spf13/cobra.(*Command).Execute
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:902
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/cli.go:303
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithArgs.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:433
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithArgs
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:434
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:291
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithCapture.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:300
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.captureOutput
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:352
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithCapture
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:299
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestUnavailableZip.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip_test.go:441
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1197
        	            	Wraps: (2) secondary error attachment
        	            	  | runtime error: invalid memory address or nil pointer dereference
        	            	  | (1) runtime error: invalid memory address or nil pointer dereference
        	            	  | Error types: (1) runtime.errorString
        	            	Wraps: (3) panic: runtime error: invalid memory address or nil pointer dereference
        	            	Error types: (1) *withstack.withStack (2) *secondary.withSecondaryError (3) *errutil.leafError
        	Test:       	TestUnavailableZip/server_1
    --- FAIL: TestUnavailableZip/server_1 (32.05s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.TestUnavailableZip [failed](https://tanzanite.cluster.engflow.com/invocations/default/19943fca-2ba0-4c23-a6b6-53091a43f0a4?testReportRun=6&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on master @ [cc4fdffa8532d16544c48ef036689763f737dc6b](https://github.com/cockroachdb/cockroach/commits/cc4fdffa8532d16544c48ef036689763f737dc6b):


```
        	            	  | runtime.panicmem
        	            	  | 	GOROOT/src/runtime/panic.go:261
        	            	  | runtime.sigpanic
        	            	  | 	GOROOT/src/runtime/signal_unix.go:861
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.(*debugZipContext).collectPerNodeData
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip_per_node.go:476
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip.func3.2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:305
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.(*debugZipContext).forAllNodes
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:107
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip.func3
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:304
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:347
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.init.MaybeDecorateError.func106
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/clierrorplus/decorate_error.go:72
        	            	  | github.com/spf13/cobra.(*Command).execute
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:856
        	            	  | github.com/spf13/cobra.(*Command).ExecuteC
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:974
        	            	  | github.com/spf13/cobra.(*Command).Execute
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:902
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/cli.go:303
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithArgs.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:433
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithArgs
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:434
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:291
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithCapture.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:300
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.captureOutput
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:352
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithCapture
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:299
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestUnavailableZip.func3
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip_test.go:474
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) secondary error attachment
        	            	  | runtime error: invalid memory address or nil pointer dereference
        	            	  | (1) runtime error: invalid memory address or nil pointer dereference
        	            	  | Error types: (1) runtime.errorString
        	            	Wraps: (3) panic: runtime error: invalid memory address or nil pointer dereference
        	            	Error types: (1) *withstack.withStack (2) *secondary.withSecondaryError (3) *errutil.leafError
        	Test:       	TestUnavailableZip/server_2
    --- FAIL: TestUnavailableZip/server_2 (38.45s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.TestUnavailableZip [failed](https://tanzanite.cluster.engflow.com/invocations/default/9bf21f38-47fd-44ee-9ab1-67dabe38b27d?testReportRun=11&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on master @ [24547fb1115903d432004207fc49a87262133c34](https://github.com/cockroachdb/cockroach/commits/24547fb1115903d432004207fc49a87262133c34):


```
        	            	  | runtime.panicmem
        	            	  | 	GOROOT/src/runtime/panic.go:261
        	            	  | runtime.sigpanic
        	            	  | 	GOROOT/src/runtime/signal_unix.go:861
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.(*debugZipContext).collectPerNodeData
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip_per_node.go:476
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip.func3.2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:305
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.(*debugZipContext).forAllNodes
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:107
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip.func3
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:304
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.runDebugZip
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip.go:347
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.init.MaybeDecorateError.func106
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/clierrorplus/decorate_error.go:72
        	            	  | github.com/spf13/cobra.(*Command).execute
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:856
        	            	  | github.com/spf13/cobra.(*Command).ExecuteC
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:974
        	            	  | github.com/spf13/cobra.(*Command).Execute
        	            	  | 	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:902
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/cli.go:303
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithArgs.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:433
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithArgs
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:434
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:291
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithCapture.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:300
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.captureOutput
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:352
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestCLI.RunWithCapture
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/testutils.go:299
        	            	  | github.com/cockroachdb/cockroach/pkg/cli.TestUnavailableZip.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cli/zip_test.go:441
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) secondary error attachment
        	            	  | runtime error: invalid memory address or nil pointer dereference
        	            	  | (1) runtime error: invalid memory address or nil pointer dereference
        	            	  | Error types: (1) runtime.errorString
        	            	Wraps: (3) panic: runtime error: invalid memory address or nil pointer dereference
        	            	Error types: (1) *withstack.withStack (2) *secondary.withSecondaryError (3) *errutil.leafError
        	Test:       	TestUnavailableZip/server_1
    --- FAIL: TestUnavailableZip/server_1 (32.20s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=11</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cli/zip_per_node.go
+++ b/pkg/cli/zip_per_node.go
@@ -472,18 +472,18 @@ func (zc *debugZipContext) collectPerNodeData(
 				if err := zc.z.createError(sf, name, requestErr); err != nil {
 					return err
 				}
-				// Log out the list of errors that occurred during log entries request.
-				if len(entries.ParseErrors) > 0 {
-					sf.shout("%d parsing errors occurred:", len(entries.ParseErrors))
-					for _, err := range entries.ParseErrors {
-						sf.shout("%s", err)
-					}
-					parseErr := fmt.Errorf("%d errors occurred:\n%s", len(entries.ParseErrors), strings.Join(entries.ParseErrors, "\n"))
-					if err := zc.z.createError(sf, name, parseErr); err != nil {
-						return err
-					}
-				}
 				continue
+			}
+			// Log the list of errors that occurred during log entries request.
+			if len(entries.ParseErrors) > 0 {
+				sf.shout("%d parsing errors occurred:", len(entries.ParseErrors))
+				for _, err := range entries.ParseErrors {
+					sf.shout("%s", err)
+				}
+				parseErr := fmt.Errorf("%d errors occurred:\n%s", len(entries.ParseErrors), strings.Join(entries.ParseErrors, "\n"))
+				if err := zc.z.createError(sf, name, parseErr); err != nil {
+					return err
+				}
 			}
 			sf.progress("writing output: %s", name)
 			warnRedactLeak := false

